src/extension.ts

```diff
@@ -110,4 +113,89 @@ export const activate = () => {
+                return;
+            }
+
+            if (
+                document.languageId !== "json" &&
+                document.languageId !== "jsonc"
+            ) {
+                return;
+            }
+
+            if (contentChanges.length === 0) {
+                return;
+            }
+
+            const editor = vscode.window.activeTextEditor;
+
+            if (
+                document.uri !== editor?.document.uri ||
+                ["output"].includes(editor.document.uri.scheme)
+            ) {
+                return;
+            }
+            if (
+                vscode.workspace.fs.isWritableFileSystem(
+                    document.uri.scheme
+                ) === false
+            ) {
+                return;
+            }
+
+            editor.edit((edit) => {
+                contentChanges.forEach(async (content) => {
+                    if (
+                        !content.text.startsWith("\n") &&
+                        !content.text.startsWith("\r\n")
+                    ) {
+                        return;
+                    }
+
+                    const prevLinePosition = document.positionAt(content.rangeOffset);
+
+                    const prevLine = document.lineAt(prevLinePosition);
+                    const prevLineText = prevLine.text;
+
+                    const prevLineLastChar = prevLineText.at(-1);
+
+                    if (!prevLineLastChar) {
+                        return;
+                    }
+
+                    if (document.languageId === 'jsonc') {
+                        if (prevLineText.trim().startsWith("//") || prevLineText.trim().startsWith("/*")) {
+                            return;
+                        }
+
+                        const textWithouComments = stripJsonComments(document.getText());
+                        const prevLineTextWithoutComments = getTextByLine(textWithouComments, prevLine.lineNumber)?.trim();
+
+                        if (prevLineTextWithoutComments !== prevLineText.trim()) {
+                            return;
+                        }
+                    }
+
+                    const isCurrentLineEmpty =
+                        document.lineAt(prevLine.lineNumber + 1).text.trim() === "";
+
+                    const isMatchValue =
+                        prevLineLastChar === "}" ||
+                        prevLineLastChar === '"' ||
+                        prevLineLastChar === ']' ||
+                        isNumber(prevLineLastChar);
+
+                    if (isMatchValue && isCurrentLineEmpty) {
+                        // In multicursor mode last character position is broken when I use prevLinePosition, IDK why
+                        const lastCharacterPosition = new vscode.Position(prevLine.lineNumber, prevLine.range.end.character);
+                        edit.insert(lastCharacterPosition, ",");
+                    }
+                })
+            });
```

Comment:
What do you think of adding

```ts
{
    undoStopAfter: false,
    undoStopBefore: false,
}
```
to editor.edit?

Reply:
Added